README.md

```diff
@@ -534,16 +534,33 @@ order to manually set the time unit, you can specify it manually:
 BENCHMARK(BM_test)->Unit(benchmark::kMillisecond);
 ```
 
-### Reporting the mean, median and standard deviation by repeated benchmarks
-By default each benchmark is run once and that single result is reported.
+### Reporting each iteration of an run/repetition separately
+It is worth noting that internally each run/repetition usually runs several
+iterations of the benchmarked code, and then the total measurements over all
+the iterations are averaged (divided by the iteration count). It makes sense,
+since normally the measurements are expected to be noisy, and besides,
+it would be quite costly to measure each iteration separately.
+
+But, sometimes this averaging is unwelcomed.
+You can use `--benchmark_report_separate_iterations={true|false}` flag or
+`ReportSeparateIterations(bool)` function can be used to control the behaviour,
```

Comment:
i wonder if this is useful given the user can't really control the number of iterations (at least the max). Ie, this could output thousands++ of reports if the benchmark suddenly speeds up.

Reply:
That is why there is `--benchmark_report_aggregates_only=`, and i have added `--benchmark_display_aggregates_only=` :)

Reply:
Even I'm getting confused between all these options.

Can you document/explain how these all interact in a simple way?

Reply:
They don't interact **directly**. They are separate.
* `benchmark_report_aggregates_only` - for all the reporters, only output the aggregates
* `benchmark_display_aggregates_only` - for display reporter, only print the aggregates; file reporter will still contain both the non-aggregates, and aggregates.
* `benchmark_report_separate_iterations` - when benchmarking, don't do one huge measurement (spanning `n` iterations), and then divide it by iteration count. instead, do all these `n` iterations separately. I.e. do no implicit averaging of measurements.

Reply:
`--benchmark_report_aggregates_only --benchmark_report_separate_iterations` does the same as `--benchmark_report_aggregates_only` ?

Reply:
No.
If there were repetitions, the stddev aggregate may (will) be shockingly different between `--benchmark_report_aggregates_only --benchmark_report_separate_iterations` and `--benchmark_report_aggregates_only`.

Thinking about it, maybe `--benchmark_report_separate_iterations` isn't such a great name.
`report` looks out of place.. I can clearly see that it confuses you here.
It has different meaning as compared to to `benchmark_report_aggregates_only` / `benchmark_display_aggregates_only`.

Reply:
maybe just --benchmark_report_every_iteration or --benchmark_average_over_iterations (with the default being true) or --benchmark_one_iteration_per_run or maybe just --benchmark_max_iterations=1 ?

Reply:
Certainly not `--benchmark_max_iterations=1`.
It looks way too similar to `benchmark_min_time`, while having radically different meaning.

`--benchmark_report_every_iteration` still looks quite similar to the current `--benchmark_report_separate_iterations`.

So i guess `--benchmark_average_over_iterations` / `--benchmark_one_iteration_per_run`
is the closest hit.
Maybe `--benchmark_average_over_iterations`.